common/src/main/java/io/apicurio/registry/types/ArtifactState.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2019 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.types;
+
+/**
+ * @author Ales Justin
+ */
+public enum ArtifactState {
+
+    ENABLED,
+    DISABLED,
+    DEPRECATED,
+    DELETED
```

Comment:
Is this needed?  Are we actually soft-deleting artifacts in the storage layer?  If so, I think a side effect of this is that a user could delete an artifact by calling "updateArtifactState" and passing it `DELETE`.  Maybe that's ok?

Reply:
That's how I have it now ... 
I added it since it (the DELETE state) came around when we discussed it with Andrew ...